src/node_file.cc

```diff
@@ -48,6 +48,11 @@ using v8::Value;
 
 #define GET_OFFSET(a) ((a)->IsNumber() ? (a)->IntegerValue() : -1)
 
+static int SanitizeFD(Local<Value> fd) {
+  CHECK(fd->IsUint32() && "file descriptor must be a positive 32-bit integer");
```

Comment:
I thought that this would abort and when it aborts, it would point to this line and we will have better error message.